### PR TITLE
Chore | migrate source plugin to TSDX

### PR DESCRIPTION
**Pull request by tlgimenes**
## What's the purpose of this pull request?
As title says, this PR migrates source plugin to tsdx. Also, add some tests

## How to test it?
https://github.com/vtex-sites/btglobal.store/pull/648

### CODE REVIEW

File: packages/gatsby-source-vtex/package.json [hunk 1]
```diff
@@ -1,25 +1,28 @@
 {
-  "name": "@vtex/gatsby-source-vtex",
-  "version": "0.371.11",
-  "description": "Gatsby source plugin for building websites using VTEX as a data source.",
-  "main": "index.js",
-  "scripts": {
-    "build": "tsc",
-    "prepare": "cross-env NODE_ENV=production yarn run build",
-    "develop": "tsc --watch"
+  "name": "gatsby-source-vtex",
```
Comment (emersonlaurentino) [inline packages/gatsby-source-vtex/package.json:2]: why are you going to remove the prefix? `@vtex`

File: packages/gatsby-source-vtex/tsconfig.json [hunk 2]
```diff
@@ -1,8 +1,35 @@
 {
-  "extends": "../../tsconfig.json",
+  // see https://www.typescriptlang.org/tsconfig to better understand tsconfigs
```
Comment (emersonlaurentino) [inline packages/gatsby-source-vtex/tsconfig.json:2]: I don't understand why remove extends?

File: packages/gatsby-source-vtex/package.json (hunk 1 above)
Comment (tlgimenes) [reply packages/gatsby-source-vtex/package.json:2]: oops, thanks!

File: packages/gatsby-source-vtex/tsconfig.json (hunk 2 above)
Comment (tlgimenes) [reply packages/gatsby-source-vtex/tsconfig.json:2]: I've just fallback to tsdx's standard definition. But I can merge with extends

**Review by emersonlaurentino — APPROVED** [review]

**Pull request merged**